src/metatables.cpp

```diff
@@ -0,0 +1,412 @@
+    return _binary_metamethod(ctx, std::move(location), "__sub", &Value::sub);
+}
+
+auto mul(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(ctx, std::move(location), "__mul", &Value::mul);
+}
+
+auto div(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(ctx, std::move(location), "__div", &Value::div);
+}
+
+auto mod(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(ctx, std::move(location), "__mod", &Value::mod);
+}
+
+auto pow(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(ctx, std::move(location), "__pow", &Value::pow);
+}
+
+auto idiv(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(ctx, std::move(location), "__idiv", &Value::int_div);
+}
+
+auto band(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(
+        ctx, std::move(location), "__band", &Value::bit_and, "bitwise operation");
+}
+
+auto bor(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(
+        ctx, std::move(location), "__bor", &Value::bit_or, "bitwise operation");
+}
+
+auto bxor(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(
+        ctx, std::move(location), "__bxor", &Value::bit_xor, "bitwise operation");
+}
+
+auto shl(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(
+        ctx, std::move(location), "__shl", &Value::bit_shl, "bitwise operation");
+}
+
+auto shr(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(
+        ctx, std::move(location), "__shr", &Value::bit_shr, "bitwise operation");
+}
+
+auto concat(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    return _binary_metamethod(ctx, std::move(location), "__concat", &Value::concat, "concatenate");
+}
+
+static auto _force_bool(const CallResult& result) -> CallResult {
+    auto value = Vallist({result.values().get(0)});
+    return CallResult(value, result.source_change());
+}
+
+auto eq(const CallContext& ctx, std::optional<Range> location) -> CallResult {
+    auto arg1 = ctx.arguments().get(0);
+    auto arg2 = ctx.arguments().get(1);
+
+    auto result = std::visit(
+        overloaded{
+            [&ctx](const Table& arg1, const Table& arg2) -> CallResult {
+                // first check if they are trivially equal
+                if (arg1 == arg2) {
+                    return CallResult({true});
+                }
+
+                {
+                    auto meta_left = arg1.get_metamethod("__eq");
+                    if (meta_left.is_function()) {
+                        return meta_left.call(ctx);
+                    }
+                }
```

Comment:
what kind of syntax is that?

Reply:
What do you mean? It's just an `if`...

Reply:
i mean the brackets around the `if`. That confuses me because i never seen something like that

Reply:
They create a new scope/block (like lua do blocks, I guess). It looks a bit strange but is a tiny bit more correct because you can't (accidentally) use `meta_left` after the block.

Reply:
ok, thanks for the explanation.